components/script/script_thread.rs

```diff
@@ -2327,16 +2307,64 @@ impl ScriptThread {
     /// for the given pipeline (specifically the "navigate" algorithm).
     fn handle_navigate(&self, parent_pipeline_id: PipelineId,
                               browsing_context_id: Option<BrowsingContextId>,
-                              load_data: LoadData,
+                              mut load_data: LoadData,
                               replace: bool) {
         match browsing_context_id {
             Some(browsing_context_id) => {
                 let iframe = self.documents.borrow().find_iframe(parent_pipeline_id, browsing_context_id);
                 if let Some(iframe) = iframe {
                     iframe.navigate_or_reload_child_browsing_context(Some(load_data), NavigationType::Regular, replace);
                 }
+
+                // TODO: Test javascript: urls in iframes
```

Comment:
This looks more like a "note to self"; want to file an issue for this and remove the comment instead?

Reply:
I was planning on looking into it (and [this `TODO`]( https://github.com/servo/servo/pull/17083/commits/fa3e9ab2444daab2c67b7e25fd6a5f88264b4ddb#diff-1879ac6bd5d567e2aa43529d33474677R1685)) a bit more in this pull request. I'll open new issues if they aren't easy to fix!

Reply:
I added support for `<iframe src="javascript:'something'"></iframe>` in https://github.com/servo/servo/pull/17083/commits/6ae6031468662161280a4353da9c0c23368cd83d. This was necessary to prevent a regression from `master`.

I fixed the other `TODO` in https://github.com/servo/servo/pull/17083/commits/ff786a050a9b8c8f526130e4bca9b2208f7998c4.